pkg/cmd/service/service_test.go

```diff
@@ -224,6 +225,115 @@ func TestBuildServerWithPresharedKeyAuthentication(t *testing.T) {
 	require.NoError(t, service.Close(ctx))
 }
 
+func TestSettingCORS(t *testing.T) {
+	os.Setenv("OPENFGA_AUTH_METHOD", "preshared")
+	os.Setenv("OPENFGA_AUTH_PRESHARED_KEYS", "KEYONE,KEYTWO")
+	os.Setenv("OPENFGA_CORS_ALLOWED_ORIGINS", "http://openfga.dev,http://localhost")
+	os.Setenv("OPENFGA_CORS_ALLOWED_HEADERS", "Origin,Accept,Content-Type,X-Requested-With,Authorization,X-Custom-Header")
+
+	corsAllowedOrigins := GetServiceConfig().CORSAllowedOrigins
+	expectedCORSAllowedOrigins := []string{"http://openfga.dev", "http://localhost"}
+	if !reflect.DeepEqual(corsAllowedOrigins, expectedCORSAllowedOrigins) {
+		t.Fatalf("Unexpected CORSAllowedOrigin expected %v, actual %v", corsAllowedOrigins, expectedCORSAllowedOrigins)
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+
+	service, err := BuildService(GetServiceConfig(), logger.NewNoopLogger())
+	require.NoError(t, err)
+
+	g := new(errgroup.Group)
+	g.Go(func() error {
+		return service.Run(ctx)
+	})
+
+	ensureServiceUp(t)
+
+	type args struct {
+		origin string
+		header string
+	}
+	type want struct {
+		origin string
+		header string
+	}
+	tests := []struct {
+		name string
+		args args
+		want want
+	}{
+		{
+			name: "Good Origin",
+			args: args{
+				origin: "http://localhost",
+				header: "Authorization, X-Custom-Header",
+			},
+			want: want{
+				origin: "http://localhost",
+				header: "Authorization, X-Custom-Header",
+			},
+		},
+		{
+			name: "Bad Origin",
+			args: args{
+				origin: "http://openfga.example",
+				header: "X-Custom-Header",
+			},
+			want: want{
+				origin: "",
+				header: "",
+			},
+		},
+		{
+			name: "Bad Header",
+			args: args{
+				origin: "http://localhost",
+				header: "Bad-Custom-Header",
+			},
+			want: want{
+				origin: "",
+				header: "",
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			payload := strings.NewReader(`{"name": "some-store-name"}`)
+			req, err := http.NewRequest("OPTIONS", fmt.Sprintf("%s/stores", openFgaServerURL), payload)
+			require.NoError(t, err, "Failed to construct request")
+			req.Header.Set("content-type", "application/json")
+			req.Header.Set("authorization", "Bearer KEYTWO")
+			req.Header.Set("Origin", test.args.origin)
+			req.Header.Set("Access-Control-Request-Method", "OPTIONS")
+			req.Header.Set("Access-Control-Request-Headers", test.args.header)
+			retryClient := http.Client{}
+			res, err := retryClient.Do(req)
+			require.NoError(t, err, "Failed to execute request")
+
+			origin := res.Header.Get("Access-Control-Allow-Origin")
+			acceptedHeader := res.Header.Get("Access-Control-Allow-Headers")
+			if origin != test.want.origin {
```

Comment:
Can use `require.Equal` here, and elsewhere, if you would like. I think it is a bit cleaner.